content/cucumber/browser-automation.md

```diff
@@ -16,86 +17,149 @@ Automation tools such as:
-        // Enter something to search for
+   @When("^I search for \"(.*)\"$")
+     public void search_for(String query) {
+        WebElement element = browser.findElelment(By.name("q"));
+        \\Enter Something to search for
         element.sendKeys(query);
-        // Now submit the form. WebDriver will find the form for us from the element
+        \\Now submit the form. WebDriver will find the form for us from the element
         element.submit();
-    }
-
-    @Then("^the page title should start with \"(.*)\"$")
-    public void checkTitle() {
-        // Google's search is rendered dynamically with JavaScript.
-        // Wait for the page to load, timeout after 10 seconds
-        (new WebDriverWait(driver, 10)).until(new ExpectedCondition<Boolean>() {
-          public Boolean apply(WebDriver d) {
-          return d.getTitle().toLowerCase().startsWith("cheese");
-          }
-        });
-        assertThat(driver.getTitle(), startsWith("cheese"));
-        // Should see: "cheese! - Google Search"
+   }
+
+   @Then("^ the page title should start with \"(.*)\"$")
+   public void checkTitle() {
+       \\Google's search is rendered dynamically with JavaScript.
+       \\Wait for the page to load timeout after ten seconds
+       new WebDriverWait(driver,'10')).untill(new ExpectedCondition&lt;Boolean&gt; {
+       public Boolean apply(WebDriver d) {
+       return d.getTitle().toLowerCase.startsWith("cheese");
+       \\Should see: "cheese! -Google Search"
     }
 
     @After()
-    public void closeBrowser() {
-        driver.quit();
-    }
+     public void closeBrowser() {
+       driver.quit();
+     }
 }
 ```
 
-[/carousel]
 
-## Watir
+```ruby
+ require 'rubygems'
+ require 'selenium-webdriver'
+
+ Given("^I am on the Google search page$") do
+  driver = Selenium::WebDriver.for :firefox
+  driver.get "http:\\google.com"
+ end
+
+ When("^I search for "([^"]*)"$") do
+  element = driver.find_element(:name => "q")
+  element.send_keys "Cheese!"
+  element.submit
+ end
+
+ Then("^the page title should start with "([^"]*)"$") do
+  wait = Selenium::WebDriver::Wait.new(:timeout => 10)
+  wait.until { driver.title.downcase.start_with? "cheese!" }
+  puts "Page title is #{driver.title}"
+   browser.close
+ end
+```
+
+## Watir Webdriver
 
-TODO
+Watir, pronounced water, is an open-source (BSD) family of Ruby libraries for automating web browsers. It allows you to write tests that are easy to read and maintain. It is simple and flexible.
+
+Watir drives browsers the same way people do. It clicks links, fills in forms, presses buttons. Watir also checks results, such as whether expected text appears on the page.
+
+Watir is a family of Ruby libraries but it supports your app no matter what technology it is developed in. While Watir supports only Internet Explorer on Windows, Here comes Watir-WebDriver to solve single browser testing and support Chrome, Firefox, Internet Explorer, Opera and also running in headless mode (HTMLUnit).
```

Comment:
app -> application(?)
what -> which(?)
some text seems to be missing (abrupt change from "IE on Windows, Here comes")